sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,755 @@
+  memory_pool(context ctx, Properties props = {});
+
+  template <typename Properties = empty_properties_t>
+  memory_pool(context ctx, device dev, usm::alloc kind, Properties props = {});
+
+  ~memory_pool();
+
+  context get_context() const;
+
+  std::vector<device> get_devices() const;
+
+  usm::alloc get_alloc_kind() const;
+
+  size_t get_max_size() const;
+
+  size_t get_threshold() const;
+
+  void set_new_threshold(size_t newThreshold);
+
+}; // memory_pool
+
+}  // ext::oneapi::experimental
+----
+
+[source, c++]
+----
+template <typename Properties = empty_properties_t>
+memory_pool(context ctx, Properties props = {});
+----
+
+_Effects_: Constructs a memory pool associated with `ctx` and all SYCL devices
+associated with it, with the allocation kind `usm::host` and applying any
+properties in `props`.
+
+[source, c++]
+----
+template <typename Properties = empty_properties_t>
+memory_pool(context ctx, device dev, usm::alloc kind, Properties props = {});
+----
+
+_Effects_: Constructs a memory pool associated with `ctx` and `dev`, with the
+allocation kind `kind` and applying any properties in `props`.
+
+_Throws_: An exception with the `errc::invalid` error code if `kind` is
+`usm::host`.
+
+[source, c++]
+----
+~memory_pool();
+----
+
+_Effects_: If this was the last copy, signals to the SYCL runtime for the memory
+pool to be destroyed after all remaining allocations have been freed, and
+returns immediately without waiting.
+
+[source, c++]
+----
+context get_context() const;
+----
+
+_Returns_: The SYCL context associated with the memory pool.
+
+[source, c++]
+----
+std::vector<device> get_devices() const;
+----
+
+_Returns_: The SYCL device(s) associated with the memory pool.
+
+[source, c++]
+----
+usm::alloc get_alloc_kind() const;
+----
+
+_Returns_: The memory allocation kind of the memory pool.
+
+[source, c++]
+----
+size_t get_max_size() const;
+----
+
+_Returns_: The maximum size of the memory pool.
+
+[source, c++]
+----
+size_t get_threshold() const;
+----
+
+_Returns_: The deallocation threshold of the memory pool.
+
+[source, c++]
+----
+void set_new_threshold(size_t newThreshold);
+----
+
+_Effects_: Sets the deallocation threshold of the memory pool if the value of
+`newThreshold` is larger than the current threshold.
+
+_Throws_: An exception with the `errc::invalid` error code if the value of
+`newThreshold` is lower than the current threshold or larger than the maximum
+allocation size.
+
+
+=== Memory pool properties
+
+A memory pool can be constructed with a number of properties which can change
+certain behaviors, these can be specified when constructing a `memory_pool`
+object.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+struct initial_threshold_key {
+  using value_t = property_value<initial_threshold_key>;
+
+  initial_threshold_key(size_t initialThreshold);
+};
+
+struct maximum_size_key {
+  using value_t = property_value<maximum_size_key>;
+
+  initial_threshold_key(size_t maxSize);
```

Comment:
```suggestion
  maximum_size_key(size_t maxSize);
```

Reply:
Fixed in https://github.com/intel/llvm/commit/48493ace0babceba6b65fe7399b43284c10681db.